packages/react-ai/src/components/AIConversation/views/Controls/FieldControl.tsx

```diff
@@ -72,23 +73,25 @@ const TextInput: typeof TextAreaBase = React.forwardRef(
     const isFirstMessage = !messages || messages.length === 0;
 
     React.useEffect(() => {
-      const textarea = document.getElementById(`${FIELD_BLOCK}-text-input`);
-
       const handleResize = () => {
-        if (textarea) {
-          textarea.style.height = 'auto';
-          textarea.style.height = `${textarea.scrollHeight}px`;
+        if (textAreaRef && textAreaRef.current) {
+          textAreaRef.current.style.height = 'auto';
+          textAreaRef.current.style.height = `${textAreaRef.current.scrollHeight}px`;
```

Comment:
nit: Might be nice to put this in a custom hook?

Reply:
Moved to a custom hook!